nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -1064,6 +1111,21 @@ in {
           # "The files of the app [all apps in /var/lib/nextcloud/apps] were not replaced correctly"
           # Restarting phpfpm on Nextcloud package update fixes these issues (but this is a workaround).
           restartTriggers = [ webroot overrideConfig ];
+        } // lib.optionalAttrs requiresRuntimeSystemdCredentials {
+          serviceConfig.LoadCredential = runtimeSystemdCredentials;
+
+          # FIXME: We use a hack to make the credential files readable by the nextcloud
+          #        user by copying them somewhere else and overriding CREDENTIALS_DIRECTORY
+          #        for php. This is currently necessary as the unit runs as root.
+          preStart = ''
+            umask 0077
+            # NOTE: The phpfpm runtime directory is currently preserved
+            #       between restarts.
+            rm -rf /run/phpfpm/nextcloud-credentials/
```

Comment:
Also... doesn't this mean that other phpfpm instances can see these values (even though only readable as root, so not from PHP code)?

Reply:
> I think it would be nicer to use `$RUNTIME_DIRECTORY/nextcloud-credentials/` here.

I don't think this currently applies here but `$RUNTIME_DIRECTORY` has the pesky detail that you can technically specify multiple directories which will end up as `/run/a:/run/b` in the variable

Reply:
> Also... doesn't this mean that other phpfpm instances can see these values (even though only readable as root, so not from PHP code)?

Yes sadly thats the case, briefly as `root` and then as `nextcloud`. I don't know phpfpm good enough to know what would happen if the service were rootless, but that would be much cleaner here

Reply:
> I don't think this currently applies here but $RUNTIME_DIRECTORY has the pesky detail that you can technically specify multiple directories which will end up as /run/a:/run/b in the variable

Meh, makes perfect sense for a variable name in signular :/ 
but good point, I'll retract that.

> Yes sadly thats the case, briefly as root and then as nextcloud. I don't know phpfpm good enough to know what would happen if the service were rootless, but that would be much cleaner here

My main problem is that the runtime directory is used for each phpfpm service, i.e. each PHP application on the same host will have this in its runtime directory (sure, different user then, but this smells like a disaster waiting to happen).

But given that we can specify multiple runtime directories, wouldn't it be an option to add another directory to the phpfpm-nextcloud service and place the secrets into that dir?

Reply:
> But given that we can specify multiple runtime directories, wouldn't it be an option to add another directory to the phpfpm-nextcloud service and place the secrets into that dir?

Sure, good point. I'll work on this in a bit